packages/app-root/src/app/groups/[groupId]/GroupStatsContainer.js

```diff
@@ -35,45 +49,30 @@ function GroupStatsContainer({
 
   useEffect(function updateStartDateParam() {
     if (selectedStartDate && (startDate === undefined)) {
-      updateQueryParams([['start_date', selectedStartDate]])
-    }
-  }, [selectedStartDate, startDate])
-
-  function updateQueryParams(newQueryParams) {
-    const queryParams = new URLSearchParams(window.location.search)
-
-    for (const [key, value] of newQueryParams) {
-      if (!value) {
-        queryParams.delete(key);
-      } else {
-        queryParams.set(key, value);
-      }
+      const newQueryParams = updateQueryParams([['start_date', selectedStartDate]])
+      router.push(`${window.location.pathname}?${newQueryParams.toString()}`)
     }
-
-    router.push(`${window.location.pathname}?${queryParams.toString()}`)
-  }
+  }, [selectedStartDate, startDate, router])
```

Comment:
The missing dependency, that's generating the original error, is the client-side router here.